compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/FirHelpers.kt

```diff
@@ -268,7 +269,7 @@ fun FirMemberDeclaration.implicitModality(context: CheckerContext): KtModifierKe
 private fun FirDeclaration.modifierListOrNull() = (this.source.getModifierList() as? FirPsiModifierList)?.modifierList
 
 private fun FirDeclaration.hasBody(): Boolean = when (this) {
-    is FirSimpleFunction -> this.body != null && this.body !is FirSingleExpressionBlock
-    is FirProperty -> this.setter != null || this.getter != null
+    is FirSimpleFunction -> this.body != null && this.body !is FirEmptyExpressionBlock
```

Comment:
I think this is a mistake. This declaration

```
fun foo() {}
```

still has body, but this one

```
abstract fun foo()
```

has no body.

Please make sure that `fun foo() {}` is handled properly by your checker.

Reply:
I checked it, this is ok. I added some tests for this case